### PR TITLE
[JENKINS-47896] SerializableOnlyOverRemoting

### PR DESCRIPTION
**Pull request by jglick**
#3144 with a merge conflict resolved and a minor fixup. See https://github.com/jenkinsci/remoting/pull/206 for background.

# Proposed Changelog

* RFE (internal), Prevent occasional serialization of Remoting classes to a wrong channel

### CODE REVIEW

**Review by batmat — APPROVED** [review]

**Comment by daniel-beck** [comment]
Someone else needs to write a changelog for this.

**Comment by jglick** [comment]
No changelog is needed here.

**Review by oleg-nenashev — APPROVED** [review]
should be OK if I recall the change correctly

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 1]
```diff
@@ -1828,7 +1830,7 @@ void skip0() {
     /**
      * Represents a process tree over a channel.
      */
-    public static class Remote extends ProcessTree implements Serializable {
+    public static class Remote extends ProcessTree {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/util/ProcessTree.java]: Could it be a breaking change? I know it comes from https://github.com/jenkinsci/jenkins/pull/3144 , but I am not sure why I did that

Comment (jglick) [reply core/src/main/java/hudson/util/ProcessTree.java]: This `implements Serializable` clause was always redundant, given the superclass:

```java
public abstract class ProcessTree implements Iterable<OSProcess>, IProcessTree, Serializable
```

Comment (oleg-nenashev) [reply core/src/main/java/hudson/util/ProcessTree.java]: Right. One month without coding for me :D

**Comment by oleg-nenashev** [comment]
FYI https://issues.jenkins-ci.org/browse/JENKINS-57229 is most likely caused by this PR

**Comment by oleg-nenashev** [comment]
Also Maven Plugin and SCoverage according to https://issues.jenkins-ci.org/browse/JENKINS-57244 . So we have 4 plugins which misuse serialization

The proposed plan is to...

* Disable the serialization failure by default
* Add Telemetry as @daniel-beck suggests + Add system logging

**Pull request merged**